pkg/agent/agent.go

```diff
@@ -85,6 +86,24 @@ func getConfigMap(ctx context.Context, c client.Client, name string) (*corev1.Co
 	return &cm, nil
 }
 
+func getNamespace(ctx context.Context, c client.Client, name string) (*corev1.Namespace, error) {
+	ns := corev1.Namespace{}
+
+	err := c.Get(ctx, client.ObjectKey{
+		Name: name,
+	}, &ns)
+	if err != nil {
+		return nil, err
+	}
+
+	ns.TypeMeta = metav1.TypeMeta{
```

Comment:
Do we really need TypeMeta?

Reply:
as of now, we are interested in only the uid of namespace.